pyqtgraph/styleRegistry.py

```diff
@@ -0,0 +1,369 @@
+from .Qt import QtCore, QtGui
+
+import numpy as np
+import weakref
+import itertools
+import collections
+import warnings
+import time
+
+__all__ = ['StyleRegistry']
+
+DEFAULT_COLORS = {
+    'b': QtGui.QColor(  0,  0,255,255),
+    'g': QtGui.QColor(  0,255,  0,255),
+    'r': QtGui.QColor(255,  0,  0,255),
+    'c': QtGui.QColor(  0,255,255,255),
+    'm': QtGui.QColor(255,  0,255,255),
+    'y': QtGui.QColor(255,255,  0,255),
+    'k': QtGui.QColor(  0,  0,  0,255),
+    'w': QtGui.QColor(255,255,255,255),
+    's': QtGui.QColor(100,100,150,255),
+    'gr_acc':QtGui.QColor(200,200,100,255), # graphical accent color: pastel yellow
+    'gr_reg':QtGui.QColor(  0,  0,255, 50)  # graphical region marker: translucent blue
+}
+
+# build gray values m0 to m9 to be roughly linear in CIE lab lightness:
+DEFAULT_RAMP = {
+    'm{:d}'.format(idx): QtGui.QColor(val, val, val, 255)
+    for idx, val in enumerate(
+        (0x00, 0x1e, 0x35, 0x4e, 0x69, 0x85, 0xa2, 0xc0, 0xdf, 0xff)
+    ) }
+DEFAULT_COLORS.update(DEFAULT_RAMP)
+
+# take gray values from ramp:
+DEFAULT_COLORS['d'] = DEFAULT_COLORS['m5']
+DEFAULT_COLORS['l'] = DEFAULT_COLORS['m7']
+
+for idx, col in enumerate( ( # twelve predefined plot colors
+    'l','y','r','m','b','c','g','d'
+) ): 
+    key = 'p{:X}'.format(idx)
+    DEFAULT_COLORS[key] = DEFAULT_COLORS[col]
+
+for key, col in [ # add functional colors
+    ('gr_fg','d'),  # graphical foreground
+    ('gr_bg','k'),  # graphical background
+    ('gr_txt','d'), # graphical text color
+    ('gr_hlt','r')  # graphical hover color
+]:
+    DEFAULT_COLORS[key] = DEFAULT_COLORS[col]
+
+del key, col # let the linter know that we are done with these
+
+def _expand_rgba_hex_string(hex):
```

Comment:
This behavior (or very similar) is already implemented inside `pyqtgraph.functions.mkColor`, so we should try to refactor that out and share it.